### DISCUSSION TITLE
Pore classification

### DISCUSSION THREAD

**Discussion by bbe020** [PMEAL/porespy#785] [Q&A]
Hi,
I have a binary image of grains and a connected pore. and I was wondering if it's possible to classify the pores as pore throats and pores using the PoreSpy ?

**Comment by jgostick** [comment]
Definitely, but the best approach depends on what you want to do with the result. You can extract a pore network usin ``porespy.networks.snow2``, then import it into OpenPNM. You can view OpenPNM networks in Paraview, and even overlay the original image and the network.  Here is an [example](https://porespy.org/examples/networks/reference/snow2.html) outline how it works.  If you just want to divide the image into pores and throats, then maybe ``porespy.filters.local_thickness`` would be of interest.  This will draw spheres in the image so idenfies pore bodies, though not throats.  If you let me know more specifically what you're looking for I can suggest something.

**Reply by bbe020** [reply]
Hi,
Thanks for the suggestion. I want something more like the code described in this paper:
 "Wei, B., Hou, J., Lei, Z. et al. A hybrid method to segment the pores and throats of micromodels. Arab J Geosci 12, 716 (2019). https://doi.org/10.1007/s12517-019-4925-8".

I'am running a foam experiment on micromodel, and I want to count the appearance of bubbles in each pore.
I don't think that porespy.filters.local_thickness will work for that purpose, as it doesn't define the pores properly.

The porespy.networks.snow2 can work, but it divides many pores to two or three pores.

**Reply by jgostick** [reply]
The definition of where one pore stops and another one begins is quite tricky.  The snow function uses a watershed, and actually tries to *reduce* the amount of oversegmentation.  You can adjust how much oversegmentation you get by changing the sigma and r_max arguments maybe.  Check out the detailed example of calling the individual steps in snow [here](https://porespy.org/examples/networks/tutorials/snow_advanced.html).  You can also [trim nearby peaks](https://porespy.org/examples/filters/reference/trim_nearby_peaks.html) which might reduce the splitting of pores further.
